Source/ACE.Entity/Character.cs

```diff
@@ -178,6 +181,27 @@ private void AddSkill(Skill skill, SkillStatus status, uint ranks)
         }
 
         /// <summary>
+        /// Returns a read only list of the characters friends
+        /// </summary>
+        public ReadOnlyCollection<Friend> GetFriends()
+        {
+            return new ReadOnlyCollection<Friend>(friends);
```

Comment:
This is going to cause the full recreation of the collection every time the property is hit.  You need to cache the read-only collection and reset it any time the base friends collection is reassigned.